packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -0,0 +1,469 @@
+import type { ChildProp, FlightSegmentPath } from './types'
+import React from 'react'
+import { isClientReference } from '../../lib/client-reference'
+import { getLayoutOrPageModule } from '../lib/app-dir-module'
+import type { LoaderTree } from '../lib/app-dir-module'
+import { interopDefault } from './interop-default'
+import { preloadComponent } from './preload-component'
+import { addSearchParamsIfPageSegment } from './create-flight-router-state-from-loader-tree'
+import { parseLoaderTree } from './parse-loader-tree'
+import {
+  type CreateSegmentPath,
+  type AppRenderContext,
+  getLayerAssets,
+  createComponentAndStyles,
+  hasLoadingComponentInTree,
+} from './app-render'
```

Comment:
these imports are imported from `app-render`, then used in `create-component-tree`, then `app-render` will import `create-component-tree` again to form a circular dependency. better to move them here or create an extra file for these ones.